Ah; looks like a test needs to be moved;

```
=== Failed
=== FAIL: amd64.integration.daemon TestDaemonProxy/reload_sanitized (10.55s)
    daemon_test.go:443: timeout hit after 10s: waiting for daemon logs match
    --- FAIL: TestDaemonProxy/reload_sanitized (10.55s)

=== FAIL: amd64.integration.daemon TestDaemonProxy (0.50s)
```

--- a/cmd/dockerd/daemon.go
+++ b/cmd/dockerd/daemon.go
@@ -3,6 +3,7 @@ package main
 import (
 	"context"
 	"crypto/tls"
+	"encoding/json"
 	"fmt"
 	"net"
 	"net/http"
@@ -469,14 +470,15 @@ type builderOptions struct {
 
 func (cli *daemonCLI) reloadConfig() {
 	ctx := context.TODO()
+	log.G(ctx).WithField("config-file", *cli.configFile).Info("Got signal to reload configuration")
 	reload := func(c *config.Config) {
 		if err := validateAuthzPlugins(c.AuthorizationPlugins, cli.d.PluginStore); err != nil {
-			log.G(ctx).Fatalf("Error validating authorization plugin: %v", err)
+			log.G(ctx).WithError(err).Fatal("Error validating authorization plugin")
 			return
 		}
 
 		if err := cli.d.Reload(c); err != nil {
-			log.G(ctx).Errorf("Error reconfiguring the daemon: %v", err)
+			log.G(ctx).WithError(err).Error("Error reconfiguring the daemon")
 			return
 		}
 
@@ -497,7 +499,17 @@ func (cli *daemonCLI) reloadConfig() {
 	}
 
 	if err := config.Reload(*cli.configFile, cli.flags, reload); err != nil {
-		log.G(ctx).Error(err)
+		log.G(ctx).WithError(err).Error("Error reloading configuration")
+		return
+	}
+
+	sanitizedConfig := config.Sanitize(cli.d.Config())
+	jsonData, err := json.Marshal(sanitizedConfig)
+	if err != nil {
+		log.G(context.TODO()).WithError(err).Warn("Error when marshaling configuration for printing")
+		log.G(context.TODO()).Info("Reloaded configuration")
+	} else {
+		log.G(context.TODO()).WithField("config", string(jsonData)).Info("Reloaded configuration")
 	}
 }
 

--- a/daemon/config/config.go
+++ b/daemon/config/config.go
@@ -2,7 +2,6 @@ package config // import "github.com/docker/docker/daemon/config"
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	stderrors "errors"
 	"fmt"
@@ -380,7 +379,6 @@ func GetConflictFreeLabels(labels []string) ([]string, error) {
 
 // Reload reads the configuration in the host and reloads the daemon and server.
 func Reload(configFile string, flags *pflag.FlagSet, reload func(*Config)) error {
-	log.G(context.TODO()).Infof("Got signal to reload configuration, reloading from: %s", configFile)
 	newConfig, err := getConflictFreeConfiguration(configFile, flags)
 	if err != nil {
 		if flags.Changed("config-file") || !os.IsNotExist(err) {
@@ -800,4 +798,15 @@ func migrateHostGatewayIP(config *Config) {
 		config.HostGatewayIPs = []netip.Addr{addr}
 		config.HostGatewayIP = nil //nolint:staticcheck // ignore SA1019: clearing old value.
 	}
+}
+
+// Sanitize sanitizes the config for printing. It is currently limited to
+// masking usernames and passwords from Proxy URLs.
+func Sanitize(cfg Config) Config {
+	cfg.CommonConfig.Proxies = Proxies{
+		HTTPProxy:  MaskCredentials(cfg.HTTPProxy),
+		HTTPSProxy: MaskCredentials(cfg.HTTPSProxy),
+		NoProxy:    MaskCredentials(cfg.NoProxy),
+	}
+	return cfg
 }

--- a/daemon/config/config_test.go
+++ b/daemon/config/config_test.go
@@ -794,3 +794,26 @@ func TestMaskURLCredentials(t *testing.T) {
 		assert.Equal(t, maskedURL, test.maskedURL)
 	}
 }
+
+func TestSanitize(t *testing.T) {
+	const (
+		userPass    = "myuser:mypassword@"
+		proxyRawURL = "https://" + userPass + "example.org"
+		proxyURL    = "https://xxxxx:xxxxx@example.org"
+	)
+	sanitizedCfg := Sanitize(Config{
+		CommonConfig: CommonConfig{
+			Proxies: Proxies{
+				HTTPProxy:  proxyRawURL,
+				HTTPSProxy: proxyRawURL,
+				NoProxy:    proxyRawURL,
+			},
+		},
+	})
+	expectedProxies := Proxies{
+		HTTPProxy:  proxyURL,
+		HTTPSProxy: proxyURL,
+		NoProxy:    proxyURL,
+	}
+	assert.Check(t, is.DeepEqual(sanitizedCfg.Proxies, expectedProxies))
+}

--- a/daemon/reload.go
+++ b/daemon/reload.go
@@ -96,6 +96,7 @@ func (daemon *Daemon) Reload(conf *config.Config) error {
 
 	var txn reloadTxn
 	for _, reload := range []func(txn *reloadTxn, newCfg *configStore, conf *config.Config, attributes map[string]string) error{
+		// TODO(thaJeztah): most of these are defined as method, but don't use the daemon receiver; consider making them regular functions.
 		daemon.reloadPlatform,
 		daemon.reloadDebug,
 		daemon.reloadMaxConcurrentDownloadsAndUploads,
@@ -115,24 +116,6 @@ func (daemon *Daemon) Reload(conf *config.Config) error {
 		}
 	}
 
-	redactedConfig := struct {
-		*config.Config
-		config.Proxies `json:"proxies"`
-	}{
-		Config: &newCfg.Config,
-		Proxies: config.Proxies{
-			HTTPProxy:  config.MaskCredentials(newCfg.HTTPProxy),
-			HTTPSProxy: config.MaskCredentials(newCfg.HTTPSProxy),
-			NoProxy:    config.MaskCredentials(newCfg.NoProxy),
-		},
-	}
-	jsonData, err := json.Marshal(&redactedConfig)
-	if err != nil {
-		log.G(context.TODO()).WithError(err).Warn("Error when marshaling configuration for printing")
-		log.G(context.TODO()).Info("Reloaded configuration")
-	} else {
-		log.G(context.TODO()).Infof("Reloaded configuration: %s", jsonData)
-	}
 	daemon.configStore.Store(newCfg)
 	daemon.LogDaemonEventWithAttributes(events.ActionReload, attributes)
 	return txn.Commit()

--- a/integration/daemon/daemon_test.go
+++ b/integration/daemon/daemon_test.go
@@ -440,7 +440,7 @@ func TestDaemonProxy(t *testing.T) {
 		err := d.Signal(syscall.SIGHUP)
 		assert.NilError(t, err)
 
-		poll.WaitOn(t, d.PollCheckLogs(ctx, daemon.ScanLogsMatchAll("Reloaded configuration:", proxyURL)))
+		poll.WaitOn(t, d.PollCheckLogs(ctx, daemon.ScanLogsMatchAll("Reloaded configuration", proxyURL)))
 
 		ok, logs := d.ScanLogsT(ctx, t, daemon.ScanLogsMatchString(userPass))
 		assert.Assert(t, !ok, "logs should not contain the non-sanitized proxy URL: %s", logs)

